src/Stratis.Bitcoin.Features.Miner/PosMinting.cs

```diff
@@ -1103,9 +1103,8 @@ private int GetDepthInMainChain(UtxoStakeDescription utxoStakeDescription)
             ChainedHeader chainedBlock = this.chain.GetBlock(utxoStakeDescription.HashBlock);
 
             if (chainedBlock == null)
-                return -1;
+                return this.mempool.Exists(utxoStakeDescription.HashBlock) ? 0 : -1;
```

Comment:
mempool.Exists - Whether the transaction hash exists in the memory pool.
utxoStakeDescription.HashBlock - Block's hash.

You are checking if a tx with a hash that is equal to block's hash is in the mempool. 
It will always return `false`

Reply:
Good spot. Fixed.